### PR TITLE
Fix a NPE and provide proper cause of error.

### PR DESCRIPTION
**Pull request by ctron**
Currently, when no marshaller/unmarshaller is found, the code simply throws a NPE. In addition it will wrap all root causes into a generic INTERNAL_ERROR.

This change now throws a proper exception (ENCODER_ERROR) and provides a helpful error message.

### CODE REVIEW

**Comment by ctron** [comment]
@dwoodard1 @cdealti @MMaiero could someone of your review this, I think it is a small patch, so it really shouldn't require much time.

File: kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java [hunk 1]
```diff
@@ -1089,11 +1089,13 @@ private void writeSnapshot(long sid, XmlComponentConfigurations conf) throws Kur
         String xmlResult;
         try {
             xmlResult = marshal(conf);
-            if (xmlResult.trim().isEmpty()) {
+            if (xmlResult == null || xmlResult.trim().isEmpty()) {
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java:1092]: Am I wrong or having now marshaller throwing an exception, if result is null, makes this check for null redundant?

**Review by MMaiero — APPROVED** [review]
Hi Jens, sorry for the delay but we had few days where we were not at the office. The PR seems ok, I made only one comment.

File: kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java (hunk 1 above)
Comment (ctron) [reply kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java:1092]: It is indeed. But I would keep it in for stability. Having and additional null check prevents you from unforeseen things in the future.

Comment (MMaiero) [reply kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java:1092]: Agreed

**Pull request merged**